### PR TITLE
Fix: image contrast adjustment in RGB data plot

### PR DESCRIPTION
**Pull request by dmgav**
The pull request includes two fixes (requested by @andrewmkiss:
- Image contrast adjustment in RGB data plot is functioning properly now: the presented range of values (min and max) is selected for each color (R, G and B) using sliders; the presented data is scaled so that values within the range fill the whole dynamic range of the RGB plot (the range int 0..255 is represented as float 0..1.0). The values that fall outside the range are moved to one of the dynamic range boundaries (set equal 0.0 or 1.0). This operation does not keep quantitative information, but allows to substantially improve the visual presentation of data if its values are in narrow range or the data contain visible background noise.
- A button 'Deselect All' is added to Image Selection tool in 2D Plot tab. The logic of operation of 'Select All' and 'Deselect All' buttons is more appropriate now.

### CODE REVIEW

**Comment by dmgav** [comment]
Additional commits: improvement of Image Wizard (Element Map tab) visual representation and consistency of behavior.

File: pyxrf/model/draw_image.py [hunk 1]
```diff
@@ -267,6 +294,30 @@ def update_img_wizard_items(self):
         self.items_in_selected_group = []
         self.items_in_selected_group = list(self.dict_to_plot.keys())
 
+    def format_img_wizard_limit(self, value):
+        """
+        This function is used for formatting of range values in 'Image Wizard'.
+        The presentation of the number was tweaked so that it is nicely formatted
+           in the enaml field with adequate precision.
+        NOTE: The function is called externally from 'enaml' code.
```
Comment (mrakitin) [inline pyxrf/model/draw_image.py]:
```suggestion

        ..note::
        
        The function is called externally from 'enaml' code.
```

File: pyxrf/model/draw_image.py [hunk 2]
```diff
@@ -303,26 +353,21 @@ def set_stat_for_all(self, bool_val=False):
...
-            lowv = np.min(data_dict)
-            highv = np.max(data_dict)
-            self.range_dict[k] = {'low':lowv, 'low_defualt':lowv,
-                                  'high':highv, 'high_defualt':highv}
+        for data_name in self.dict_to_plot.keys():
+            data_arr = _normalize_data_array(self.dict_to_plot[data_name], self.scaler_data,
+                                             data_name=data_name, name_not_scalable=self.name_not_scalable)
+            lowv = np.min(data_arr)
+            highv = np.max(data_arr)
+            self.range_dict[data_name] = {'low':lowv, 'low_default':lowv,
```
Comment (mrakitin) [inline pyxrf/model/draw_image.py]:
```suggestion
            self.range_dict[data_name] = {'low': lowv, 'low_default': lowv,
```

File: pyxrf/model/draw_image.py [hunk 3]
```diff
@@ -303,26 +353,21 @@ def set_stat_for_all(self, bool_val=False):
...
+                                          'high':highv, 'high_default':highv}
```
Comment (mrakitin) [inline pyxrf/model/draw_image.py]:
```suggestion
                                          'high': highv, 'high_default': highv}
```

File: pyxrf/model/draw_image_rgb.py [hunk 4]
```diff
@@ -380,40 +380,52 @@ def _compute_equal_axes_ranges(x_min, x_max, y_min, y_max):
...
         data_b = selected_data[self.index_blue,:,:]
 
-        rgb_dict = {'data0':{'low':self.r_low, 'high':self.r_high},
-                    'data1':{'low':self.g_low, 'high':self.g_high},
-                    'data2':{'low':self.b_low, 'high':self.b_high}}
-        # norm data
-        data_r = norm_data(data_r)
-        data_g = norm_data(data_g)
-        data_b = norm_data(data_b)
+        rgb_l_h = ({'low':self.r_low, 'high':self.r_high},
```
Comment (mrakitin) [inline pyxrf/model/draw_image_rgb.py]:
```suggestion
        rgb_l_h = ({'low': self.r_low, 'high': self.r_high},
```

File: pyxrf/model/draw_image_rgb.py [hunk 5]
```diff
@@ -380,40 +380,52 @@ def _compute_equal_axes_ranges(x_min, x_max, y_min, y_max):
...
+                   {'low':self.g_low, 'high':self.g_high},
```
Comment (mrakitin) [inline pyxrf/model/draw_image_rgb.py]:
```suggestion
                   {'low': self.g_low, 'high': self.g_high},
```

File: pyxrf/model/draw_image_rgb.py [hunk 6]
```diff
@@ -380,40 +380,52 @@ def _compute_equal_axes_ranges(x_min, x_max, y_min, y_max):
...
+                   {'low':self.b_low, 'high':self.b_high})
```
Comment (mrakitin) [inline pyxrf/model/draw_image_rgb.py]:
```suggestion
                   {'low': self.b_low, 'high': self.b_high})
```

File: pyxrf/model/draw_image_rgb.py [hunk 7]
```diff
@@ -380,40 +380,52 @@ def _compute_equal_axes_ranges(x_min, x_max, y_min, y_max):
...
+
+        def _norm_data(data):
+            """
+            Normalize data between (0,1).
```
Comment (mrakitin) [inline pyxrf/model/draw_image_rgb.py]:
```suggestion
            Normalize data between (0, 1).
```

File: pyxrf/model/draw_image_rgb.py [hunk 8]
```diff
@@ -380,40 +380,52 @@ def _compute_equal_axes_ranges(x_min, x_max, y_min, y_max):
...
+            Parameters
+            ----------
+            data : 2D array
+            """
+            return (data - np.min(data)) / (np.max(data) - np.min(data))
```
Comment (mrakitin) [inline pyxrf/model/draw_image_rgb.py]: It's worth to calculate the minimum value once, and reuse it.

File: pyxrf/model/draw_image_rgb.py [hunk 9]
```diff
@@ -380,40 +380,52 @@ def _compute_equal_axes_ranges(x_min, x_max, y_min, y_max):
...
+                return data_in
+
+            if v_high - v_low < 1:  # This should not happen in practice, but check just in case
+                v_high = v_low + 1
+
+            v_low, v_high = v_low/100.0, v_high/100.0
+            c = 1.0/(v_high-v_low)
+            data_out = (data_in - v_low) * c
+            data_out[data_out < 0.0] = 0.0
+            data_out[data_out > 1.0] = 1.0
```
Comment (mrakitin) [inline pyxrf/model/draw_image_rgb.py]: `np.clip(...)` can be used here for those 2 lines.

File: pyxrf/model/draw_image_rgb.py [hunk 10]
```diff
@@ -380,40 +380,52 @@ def _compute_equal_axes_ranges(x_min, x_max, y_min, y_max):
...
+            # 'data is already normalized, so that the values are in the range 0..1
+            # v_low, v_high are in the range 0..100
+
+            if (v_low <= 0) and (v_high >= 100):
+                return data_in
+
+            if v_high - v_low < 1:  # This should not happen in practice, but check just in case
+                v_high = v_low + 1
+
+            v_low, v_high = v_low/100.0, v_high/100.0
```
Comment (mrakitin) [inline pyxrf/model/draw_image_rgb.py]:
```suggestion
            v_low, v_high = v_low / 100.0, v_high / 100.0
```

File: pyxrf/model/draw_image_rgb.py [hunk 11]
```diff
@@ -380,40 +380,52 @@ def _compute_equal_axes_ranges(x_min, x_max, y_min, y_max):
...
+            c = 1.0/(v_high-v_low)
```
Comment (mrakitin) [inline pyxrf/model/draw_image_rgb.py]:
```suggestion
            c = 1.0 / (v_high - v_low)
```

File: pyxrf/model/draw_image_rgb.py [hunk 12]
```diff
@@ -380,40 +380,52 @@ def _compute_equal_axes_ranges(x_min, x_max, y_min, y_max):
...
+            data_out[data_out > 1.0] = 1.0
+
+            return data_out
+
+        # Normalize data
+        data_r = _norm_data(data_r)
+        data_g = _norm_data(data_g)
+        data_b = _norm_data(data_b)
 
         # set limit, there should be a better way to do this
```
Comment (mrakitin) [inline pyxrf/model/draw_image_rgb.py]:
```suggestion
```

File: pyxrf/view/image2D.enaml [hunk 13]
```diff
@@ -186,19 +195,26 @@ enamldef ChooseElementAdvanced(Window): win:
...
-
-            PushButton: save_selected:
-                text = 'Save Selected as Default'
-                clicked ::
-                    img_model.record_selected()
+                                        dual_slider.low_value = 0
+                                        dual_slider.high_value = 100
+                                        #img_model.reset_low_high(loop_item)
+
+            # 'Save Selected as Default' button does not serve any useful purpose now, but it
```
Comment (mrakitin) [inline pyxrf/view/image2D.enaml]:
```suggestion
            # 'Save Selected as Default' button does not serve any useful purpose now, but
```

**Pull request merged**